### DISCUSSION TITLE
Receiving "Unathorized" response for get_messaging_actions_for_order request

### DISCUSSION THREAD

**Discussion by hh-19** [saleweaver/python-amazon-sp-api#289] [Q&A] [answered]
I am receiving the Unathorized response for the getMessagingActionsForOrder operation. I have been using multiple other endpoints without issue at all and I have the required permissions in my developer profile for BuyerCommunication and BuyerSolicitation. Below is an example of the basic request:

`Messager = Messaging(credentials=credentials)`
`res = Messager.get_messaging_actions_for_order(order_id=order_id)`

for which I receive the exception:

`sp_api.base.exceptions.SellingApiForbiddenException: [{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}]
`

My marketplace is the US, so I assume the default value for the marketplace arg but even specifying the marketplace does not solve it.  I have not had any other problems with any other endpoint so I am not sure what the issue is.

Also, thanks a lot for your work on this library @saleweaver, it has been really useful and very easy to use without any issues until now.

Thanks

**Comment by saleweaver** [comment, marked as answer]
Your app needs the permissions as well, are they set? 
If you added permissions to your app, you have to regenerate the refresh token.

**Reply by hh-19** [reply]
Thanks a lot, must have forgotten to update app permissions as well.

**Reply by saleweaver** [reply]
Glad you have it working, that's frustrating about the SP-API...
